python/caffe/test/test_net.py

```diff
@@ -72,10 +72,235 @@ def test_save_and_read(self):
         f.close()
         self.net.save(f.name)
         net_file = simple_net_file(self.num_output)
-        net2 = caffe.Net(net_file, f.name, caffe.TRAIN)
+        # Test legacy constructor
+        #   should print deprecation warning
+        caffe.Net(net_file, f.name, caffe.TRAIN)
+        # Test named constructor
+        net2 = caffe.Net(net_file, caffe.TRAIN, weights_file=f.name)
         os.remove(net_file)
         os.remove(f.name)
         for name in self.net.params:
             for i in range(len(self.net.params[name])):
                 self.assertEqual(abs(self.net.params[name][i].data
                     - net2.params[name][i].data).sum(), 0)
+
+class TestLevels(unittest.TestCase):
```

Comment:
There should be corresponding tests on the core, C++ side since not everything is in Python (though I use plenty of Python myself). Perhaps most of the tests should actually be core tests and then more abbreviated tests could be done through pycaffe.